tools/trainerproc/main.c

```diff
@@ -0,0 +1,1965 @@
+
+    for (int i = 0; literal_species[i].literal; i++)
+    {
+        if (is_literal_string(s, literal_species[i].literal))
+        {
+            fprintf(f, "%s", literal_species[i].constant);
+            return;
+        }
+    }
+
+    if (s.string_n == 0)
+    {
+        fprintf(f, "NONE");
+    }
+    else if (is_literal_string(s, "Nidoran"))
+    {
+        switch (*gender)
+        {
+        case GENDER_ANY:
+            break;
+        case GENDER_MALE:
+            fprintf(f, "NIDORAN_M");
+            break;
+        case GENDER_FEMALE:
+            fprintf(f, "NIDORAN_F");
+            break;
+        }
+        *gender = GENDER_ANY;
+    }
+    else if (is_literal_string(s, "Indeedee"))
+    {
+        switch (*gender)
+        {
+        case GENDER_ANY:
+            break;
+        case GENDER_MALE:
+            fprintf(f, "INDEEDEE_MALE");
+            break;
+        case GENDER_FEMALE:
+            fprintf(f, "INDEEDEE_FEMALE");
+            break;
+        }
+        *gender = GENDER_ANY;
+    }
+    else if (is_literal_string(s, "Meowstic"))
+    {
+        switch (*gender)
+        {
+        case GENDER_ANY:
+            break;
+        case GENDER_MALE:
+            fprintf(f, "MEOWSTIC_MALE");
+            break;
+        case GENDER_FEMALE:
+            fprintf(f, "MEOWSTIC_FEMALE");
+            break;
+        }
+        *gender = GENDER_ANY;
+    }
+    else
+    {
+        bool underscore = false;
+        static const unsigned char *male = (unsigned char *)u8"♂";
+        static const unsigned char *female = (unsigned char *)u8"♀";
+        static const unsigned char *e_diacritic = (unsigned char *)u8"é";
+        for (int i = 0; i < s.string_n; i++)
+        {
+            unsigned char c = s.string[i];
+            if ('A' <= c && c <= 'Z')
+            {
+                if (underscore)
+                    fputc('_', f);
+                underscore = false;
+                fputc(c, f);
+            }
+            else if ('a' <= c && c <= 'z')
+            {
+                if (underscore)
+                    fputc('_', f);
+                underscore = false;
+                fputc(c - 'a' + 'A', f);
+            }
+            else if (c == '\'')
+            {
+                // Do nothing.
+            }
+            else if (is_utf8_character(s, &i, male))
+            {
+                underscore = false;
+                fprintf(f, "_M");
+            }
+            else if (is_utf8_character(s, &i, female))
+            {
+                underscore = false;
+                fprintf(f, "_F");
+            }
+            else if (is_utf8_character(s, &i, e_diacritic))
+            {
+                if (underscore)
+                    fputc('_', f);
+                underscore = false;
+                fputc('E', f);
+            }
+            else
+            {
+                underscore = true;
+            }
+        }
+        if (ends_with(s, "-Alola") || ends_with(s, "-Paldea"))
```

Comment:
Doesn't seem to have support for Totem Alolan Pokémon

Reply:
With the reworked aliases in `include/constants/species.h`, this code has been removed, and so these should work :)